app/models/log_spreadsheet.rb

```diff
@@ -45,6 +44,14 @@ def self.count
     LogSpreadsheet.unscoped.count
   end
 
+  def self.remove_old_spreadsheets
+    keep = LogSpreadsheet.order(created_at: :desc).limit(SPREADSHEET_COUNT_LIMIT)
```

Comment:
To make sure this is memory optimized, you should add a select here:
`keep = LogSpreadsheet.select(:created_at).order(created_at: :desc).limit(SPREADSHEET_COUNT_LIMIT)`

Also the logic here is not obvious and a comment would be useful.  If there is only a single LogSpreadsheet in the database this keep query will include it. So you might think that it will then get deleted. But because the comparison is '<' instead of '<=' it means the last spreadsheet will be keep and nothing will be deleted.